services/explorer/src/api.rs

```diff
@@ -417,18 +418,22 @@ impl ExplorerApi {
         Self { blockchain }
     }
 
-    fn blocks(schema: Schema<&dyn Snapshot>, query: BlocksQuery) -> Result<BlocksRange, ApiError> {
+    fn blocks(schema: Schema<&dyn Snapshot>, query: BlocksQuery) -> api::Result<BlocksRange> {
         let explorer = BlockchainExplorer::from_schema(schema);
         if query.count > MAX_BLOCKS_PER_REQUEST {
-            return Err(ApiError::BadRequest(format!(
-                "Max block count per request exceeded ({})",
-                MAX_BLOCKS_PER_REQUEST
-            )));
+            return Err(api::Error::new(api::HttpStatusCode::BAD_REQUEST)
+                .title("Invalid block request")
+                .detail(format!(
+                    "Max block count per request exceeded ({})",
+                    MAX_BLOCKS_PER_REQUEST
+                )));
         }
 
         let (upper, upper_bound) = if let Some(upper) = query.latest {
             if upper > explorer.height() {
-                return Err(ApiError::NotFound(format!(
+                return Err(api::Error::new(api::HttpStatusCode::NOT_FOUND)
+                    .title("Block not found")
+                    .detail(format!(
                     "Requested latest height {} is greater than the current blockchain height {}",
```

Comment:
Nit: Weird formatting; the arguments to `format!` are not indented compared to the `format!` itself.